// angular2-jsonwebtoken/index.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require('@angular/core');
var http_1 = require("@angular/http");
var authhttp_1 = require("./services/authhttp");
var jwthelper_1 = require("./services/jwthelper");
var Angular2JWTModule = (function () {
    function Angular2JWTModule() {
    }
    Angular2JWTModule = __decorate([
        core_1.NgModule({
            imports: [http_1.HttpModule],
            providers: [authhttp_1.AuthHttp, jwthelper_1.JwtHelper]
        }), 
        __metadata('design:paramtypes', [])
    ], Angular2JWTModule);
    return Angular2JWTModule;
}());
exports.Angular2JWTModule = Angular2JWTModule;
var authhttp_2 = require("./services/authhttp");
exports.AuthHttp = authhttp_2.AuthHttp;
var jwthelper_2 = require("./services/jwthelper");
exports.JwtHelper = jwthelper_2.JwtHelper;
var authconfig_1 = require("./services/authconfig");
exports.AuthConfig = authconfig_1.AuthConfig;
